### DISCUSSION TITLE
How to read GB repro ROM

### DISCUSSION THREAD

**Discussion by rafaliyo** [sanni/cartreader#286] [General Chat]
I am trying to read a GB repro ROM, the cart reader recognise the game but when I try to read it I get:

Calculating Checksum
Result: 5935
Checksum Error

It reads the ROM but I dont know if it right

**Comment by Dakkaron** [comment]
Can you read original ROMs? Reading repros should work exactly like regular ones.

Except one thing: do you per chance use a 3D printed case for the cartreader? On my device I had too little clearance between the case and the cartridge. That then forced the pins to the side, so that they weren't aligned correctly in the reader anymore.
This only happened for repros, since their case was slightly wider. Original cartridges worked.

To test that, try the cartreader without the case. If it works fine, use a soldering iron with a tip that you don't need anymore and melt the clearance of the cartreader case a bit wider.

Another possible culprit are bad solder connections. Please just resolder all pins (especially on the slot) to make sure that you don't have any problems there.

Edit: One thing I just thought of, this will probably be what's going on:
Please open your repro and check if there is a battery in there. If there is no battery in there (and the game is still able to save) you have a hacked ROM that saves to flash instead of battery backed SRAM.

ROM dumps from one of those will probably not boot when flashed onto anything else. Dumping and flashing it to the repro will also not boot most of the time. Since they are hacked and save to flash, their checksum is always garbage. These hacks are often dirty and save corruption is frequent (e.g. random pokedex entries or Pokémon on the box are scrambled).
If you flash a regular ROM onto one of those repros, they will work fine except saving, which will not work.

To fix one of these repros, you need to add a battery and possibly a few diodes/resistors. If you post a picture of the open cartridge, then I can tell you what parts you need to add.

Once you do so, you can flash any ROM you like and saving will work.

**Reply by rafaliyo** [reply]
Thanks for the reply

I have seen although I get that fail, it extracs the ROM and if I write it, it works and save.

Yes, the repro has not battery so I assume it saves in the SRAM. I had another repro which I added a battery and it saved the game but after a few minutes it erased the save.

**Reply by Dakkaron** [reply]
If you don't have a battery, it saves to flash, not to SRAM. SRAM is volatile, meaning if it doesn't have power it deletes itself.

When adding the battery all resistors, capacitors and diodes need to be in place correctly.

Many repro boards can be either used with a flash-save-hacked ROM ir an original battery-backed SRAM ROM. When used with a hacked ROM, the manufacturer usually leaves off unnecessary components used for battery save.

If the save data is lost after a few minutes, that sounds as if the battery power doesn't reach the SRAM. I guess, you are lacking some components.

Can you send me a picture of the board?

**Comment by rafaliyo** [comment]
Here are two pics of the board (In fact are 2 boards I bought). The first one is the board as I bought it. The second one is the board with a cap, a resistor, bridges and a battery holder with its battery (I saw an indentical board on internet with those components and I soldered them
![1637604294121](https://user-images.githubusercontent.com/77173607/142913646-53db5040-e19e-497c-b1d1-ac34ffd5c59e.jpg)
)
![1637604294112](https://user-images.githubusercontent.com/77173607/142913674-ad8dc1f9-0757-4cd1-86e1-7521cce6e56b.jpg)

**Reply by Dakkaron** [reply]
The main difference I can see between your circuit and mine (I have the exact same board) is that I have diodes instead of the solder bridges on the right side. I have two BAT43 diodes with the cathode to the left side.

Also, with a battery-backed repro, make sure you don't flash battery-less-save-hacked ROMs, but original ROMs.

Btw: nice soldering! Looks much more pretty than what I did!
